hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcessSupervisor.java

```diff
@@ -196,20 +243,24 @@ private void start(Signal start) {
                 Future future = executor.execute(process);
                 logger.info("Consumer for {} was added for execution. {}", subscription.getQualifiedName(), start.getLogWithIdAndType());
 
-                runningProcesses.add(process, future);
+                consumerProcesses.add(process, future);
                 logger.info("Started consumer process for subscription {}. {}", start.getTarget(), start.getLogWithIdAndType());
             } catch (Exception ex) {
                 logger.error("Failed to create consumer for subscription {}", subscription.getQualifiedName(), ex);
             }
+        } else if (consumerProcesses.hasDyingProcess(start)) {
+            logger.info("Consumer process for {} is already dying, startup deferred.", subscription.getQualifiedName());
+	        accept(start);
```

Comment:
Process startup has to be done but can't be done when process for the same subscription is already dying. The idea is to put this signal on queue until it can be properly processed.